br/pkg/lightning/backend/external/merge_v2.go

```diff
@@ -0,0 +1,186 @@
+package external
+
+import (
+	"bytes"
+	"context"
+	"math"
+	"time"
+
+	"github.com/jfcg/sorty/v2"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/br/pkg/lightning/log"
+	"github.com/pingcap/tidb/br/pkg/membuf"
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/util/logutil"
+	"github.com/pingcap/tidb/pkg/util/size"
+	"go.uber.org/zap"
+)
+
+// MergeOverlappingFilesV2 reads from given files whose key range may overlap
```

Comment:
What is the MergeOverloapingFilesV1? Since we have not released the global sort yet if we finally use this merge implement, removing the version is ok.

Reply:
Yep, I just don't want to come up a new name for it.
Further more, I will remove one of the implementation later.